tests/Credentials/GCECredentialsTest.php

```diff
@@ -375,6 +378,26 @@ public function testGetTokenUriWithServiceAccountIdentity()
         );
     }
 
+    public function testSetIsOnGceToFalseReturnsEmptyCreds()
+    {
+        $creds = new GCECredentials();
+        $creds->setIsOnGce(false);
+        $this->assertEquals([], $creds->fetchAuthToken());
+    }
+
+    public function testSetIsOnGceToTrueWhenNotOnGceThrowsException()
+    {
+        if (GCECredentials::onGCE()) {
+            $this->markTestSkipped('Must not be on GCE for this test');
+        }
+        $this->expectException(ConnectException::class);
+
+        $httpHandler = HttpHandlerFactory::build(new Client(['timeout' => 0.1]));
```

Comment:
Would you be open to mocking out the network call instead of using a real client?